src/coreclr/jit/codegenxarch.cpp

```diff
@@ -517,7 +517,12 @@ void CodeGen::genSetRegToConst(regNumber targetReg, var_types targetType, simd_t
             }
             else if (val64.IsZero())
             {
-                emit->emitIns_SIMD_R_R_R(INS_xorps, attr, targetReg, targetReg, targetReg);
+                // Use VEX version because it's smaller (for zmm0-zmm15) than EVEX to zero a zmm register:
+                //
+                //   xorps zmm0, zmm0, zmm0 (6 bytes)
+                //   xorps xmm0, xmm0, xmm0 (4 bytes)
+                //
+                emit->emitIns_SIMD_R_R_R(INS_xorps, EA_16BYTE, targetReg, targetReg, targetReg);
```

Comment:
Might be better to use `EA_32BYTE`. Emitted size will be the same (as `EA_16BYTE`), but makes the disasm slightly better and helps differentiate any of the explicit small sizes (which frequently crop up) vs others (which are less frequent).

-- Also noting that there's no need to do this for `targetReg == XMM16-XMM31`, but I don't think we need an "extra check" for that.

Reply:
Addressed